lib/internal/fs/promises.js

```diff
@@ -121,12 +121,19 @@ async function writeFileHandle(filehandle, data, options) {
     data : Buffer.from('' + data, options.encoding || 'utf8');
   let remaining = buffer.length;
   if (remaining === 0) return;
+  // If the file is opened in APPEND mode set position as null, so that the
+  // data will be always written at the end of the file, otherwise start at 0.
+  let position = (options.flag || '').includes('a') ? null : 0;
+
   do {
     const { bytesWritten } =
       await write(filehandle, buffer, 0,
-                  Math.min(16384, buffer.length));
+                  Math.min(16384, buffer.length), position);
```

Comment:
Positional writes don't work on streams, try running this variant of writeFile to stdout.

Reply:
Agreed, if we cannot seek the position in fd, it will fail. Wouldn't that be the expected behaviour in this case?